presto-tpch/src/main/java/com/facebook/presto/tpch/TpchMetadata.java

```diff
@@ -248,57 +268,69 @@ private static ConnectorTableMetadata getTableMetadata(String schemaName, TpchTa
     @Override
     public TableStatistics getTableStatistics(ConnectorSession session, ConnectorTableHandle tableHandle, Constraint<ColumnHandle> constraint)
     {
-        TpchTableHandle table = (TpchTableHandle) tableHandle;
-        return new TableStatistics(new Estimate(getRowCount(table, Optional.of(constraint.getSummary()))), ImmutableMap.of());
+        TpchTableHandle tpchTableHandle = (TpchTableHandle) tableHandle;
+        String tableName = tpchTableHandle.getTableName();
+        TpchTable<?> tpchTable = TpchTable.getTable(tableName);
+        Predicate<Map<TpchColumn<?>, Object>> predicate = adaptPredicate(constraint);
+        TableStatisticsData tableStatisticsData = statisticsEstimator.estimateStats(tpchTable, predicate, tpchTableHandle.getScaleFactor());
+        return toTableStatistics(tableStatisticsData, tpchTableHandle, getColumnHandles(session, tpchTableHandle));
     }
 
-    private long getRowCount(TpchTableHandle tpchTableHandle, Optional<TupleDomain<ColumnHandle>> predicate)
+    private Predicate<Map<TpchColumn<?>, Object>> adaptPredicate(Constraint<ColumnHandle> constraint)
```

Comment:
`constraint` filters on columns that we are not supporting as partitioning columns. It's not consistent how we provide predicate for `TpchRecordSet`. A better approach would be to use `TupleDomain` that has only columns that are supported by Tpch connector (ORDER_STATUS). This will reduce confusion when reading the code.

Otherwise add comment why this works